enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+  - Cons
+    - user has to define the scheduling time correctly in the future
+    - gathering doesn’t happen immediately
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [roviding a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+### Open Questions
+
+- ~~Can the API define default values? How does it work or how is it implemented?~~ This is done by using `// +kubebuilder:default=`
+- Should the new time of forced gathering (if any defined by user) set a new time for the existing gathering interval? If I force gathering at time A, will the new automatic gathering be in time A + interval?
+
+### Test Plan
+
+The new config API will require updates of the existing Insights operator testsuite. The testsuite will be extended to cover the new API and respective attributes. There will be a new test checking the forced gathering feature. 
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API as the `v1alpha1` version and once we gain some confidence and experience with the new API then
+we can move it to `v1beta1`.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+The existing configuration via `support` secret should be marked as deprecated. This should be mentioned in the OCP docs (see [Support procedures](#support-procedures))
+
+### Upgrade / Downgrade Strategy
+
+If a cluster uses some non-default Insights configuration (there is an existing `support` secret overriding some default attributes) during an upgrade,
+then the config will have to be set in the new API to provide the same functionality. This is a one-time action. When there is an existings `insights.config.openshift.io` kind in a cluster before upgrade, then there is no need to update the config.
+
+### Version Skew Strategy
+
+### Operational Aspects of API Extensions
+
+This API is provided exclusively for the Insights operator and there will be no other component depending on this new API. Current configobserver
+in the Insights operator runs every 5 minutes so if we preserve the same behavior then there will be a request to the API every 5 minutes. Additionally the field
+for the forced gathering (`ForceGatheringReason` - see [Proposal](#proposal)) will require an informer.
+
+#### Failure Modes
+
+As noted in the [Risk and Mitigations](#risks-and-mitigations) section, the only possible failure is probably when the Insights operator
+cannot communicate with the config API.
```

Comment:
Now that this is becoming a formal API we often do things like validate that the newly specified remote endpoint is reachable before accepting it, I believe proxy config does this given the risk for bricking the cluster. The stakes here obviously aren't that high but it'd be nice to ensure they don't typo some config and disconnect the cluster from Insights.

Reply:
How can we ensure it? Any suggestions please?

Reply:
Essentially the spec portion of the api is viewed as a proposal, the controller continues using the status portion which will remain unchanged until the proposed new endpoint can be validated to work properly by the controller. Once the proposed endpoint is validated you update the status and the controller then uses the new endpoint. If validation fails you don't update the status and you presumably set a condition like Degraded indicating that there's an error validating the new endpoint.

This is probably overkill for at least the initial implementation.

Reply:
I see. Thanks. Well I reduced the API so now there's no endpoint to be validated.